src/app/utils/studyProgress.js

```diff
@@ -82,12 +82,10 @@ export const getStudyHistory = () => {
   return JSON.parse(localStorage.getItem(SESSIONS_KEY)) || [];
 };
 
-export const addStudyHistory = () => {
+export const addStudyHistory = date => {
   const sessionsObj = JSON.parse(localStorage.getItem(SESSIONS_KEY)) || [];
 
-  const currentDate = moment().startOf("day");
-
-  const sessions = [...sessionsObj, currentDate.format()].filter(
+  const sessions = [...sessionsObj, moment(date).format()].filter(
```

Comment:
walk me through this a little bit. So we are still using local storage -> are  we reading from local storage and then reading from the backend?

Reply:
Yeah if someone is not logged in, we store their progress/activity in the browser with local storage. If they are logged in, we read from the backend and then update the browser storage.

Here's where that happens:
https://github.com/nlaz/flashcards-for-developers/pull/23/files#diff-8e95f5b1f1768bb2f9bfa972ab63eb36R227

Reply:
We don't support every flow yet but I'm sure we will figure it out through testing.

For example, if someone starts studying then signs up, we should add save their progress to their account.

Another example, if someone is logged out and makes progress and logs back in, we should sync somehow. Currently it would ignore that logged out progress.

Reply:
Makes sense 👍 Good work